### PR TITLE
Change api endpoint used to load collection content for SavedQuestionList

### LINKED ISSUE #24660: Saved Questions data selector lists all questions for collections with the same name
**Describe the bug**
When sub-collections has the same name in several parent-collections, then the Saved Question picker will show all questions from all the sub-collections, which can be confusing.
Seems like both BE and FE needs some adjustment.

**To Reproduce**
1. Create two parent collections, "Parent1" and "Parent2"
2. Then create a sub-collection within both parents called "same-child-name"
3. Create two questions, save "Q1" to "Parent**1**/same-child-name" and save "Q2" to "Parent**2**/same-child-name"
4. New question > Saved Questions > "Parent1" > "same-child-name" - it will show both questions, since the request is for `/api/database/-1337/schema/same-child-name` without any ID or parent to make the distinction.
image

**Expected behavior**
Only show the questions related to the specific collection selected.

**Information about your Metabase Installation:**
Tested 0.41.7 thru 0.44.0

### PR DESCRIPTION
**Pull request by deniskaber**
Closes https://github.com/metabase/metabase/issues/24660

### Description

When sub-collections has the same name in several parent-collections, then the Saved Question picker will show all questions from all the sub-collections, which can be confusing.

### How to verify

To Reproduce

1. Create two parent collections, "Parent1" and "Parent2"
2. Then create a sub-collection within both parents called "same-child-name"
3. Create two questions, save "Q1" to "Parent1/same-child-name" and save "Q2" to "Parent2/same-child-name"
4. New question > Saved Questions > "Parent1" > "same-child-name" - it should show only question Q1.

### Demo

https://user-images.githubusercontent.com/7983744/228951960-e3ee5213-cafe-4e47-ae9e-70c72bc04cdb.mov

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29698)

### CODE REVIEW

**Review by WiNloSt — COMMENTED** [review]
Overall this looks good. But I want to hear some confirmation from @ranquild first if this approach is okay.

> Reviewed 7 of 8 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29698#-:-NS6wcGpBey5LGO2MhuB:bvelxbq)* status: all files reviewed, 4 unresolved discussions (waiting on @deniskaber and @ranquild)

___
*[`frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx` line 53 at r2](https://reviewable.io/reviews/metabase/metabase/29698#-NS6u6VB8kRF6DrhBNjZ:-NS6u6VB8kRF6DrhBNj_:b-4gdlz3) ([raw file](https://github.com/metabase/metabase/blob/272192811add66c0895ee774556805edb82e1f0d/frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx#L53)):*
> ```XML
>               return (
>                 <React.Fragment>
>                   {list.map(t => {
> ```

I think this new approach looks promising. But I'm not sure what `t` here is supposed to be short for. I'd probably name the list `savedQuestionsOrModels` and each item in the list as `savedquestionOrModel`, otherwise. Calling this `t` sounds too vague.

Also, ping @ranquild since this uses a different endpoint, not sure if I have missed anything. 
___
*[`frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx` line 67 at r2](https://reviewable.io/reviews/metabase/metabase/29698#-NS6upOu06RtAK795VHU:-NS6upOu06RtAK795VHV:bmnjc0h) ([raw file](https://github.com/metabase/metabase/blob/272192811add66c0895ee774556805edb82e1f0d/frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx#L67)):*
> ```XML
>                           size: 16,
>                         }}
>                         onSelect={() => onSelect({ id: virtualTableId })}
> ```

The API looks inconvenient here. What do you think about changing the passed callback API to something more straightforward e.g. `onSelect: (virtualTableId: string) => void`
https://github.com/metabase/metabase/blob/272192811add66c0895ee774556805edb82e1f0d/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L951-L958
___
*[`frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx` line 55 at r2](https://reviewable.io/reviews/metabase/metabase/29698#-NS6wC0XBIrHEL59SFTw:-NS6wC0XBIrHEL59SFTx:b-8fw2e8) ([raw file](https://github.com/metabase/metabase/blob/272192811add66c0895ee774556805edb82e1f0d/frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx#L55)):*
> ```JavaScript
> }
> 
> function mockCollectionItems() {
> ```

Should we keep the name consistent?

```ts
mockCollectionItemsEndpoint
```
___
*[`frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx` line 100 at r2](https://reviewable.io/reviews/metabase/metabase/29698#-NS6wRDx0Q0JcdxQleNY:-NS6wRDy6gky9Km8TGbG:b-ndu5wu) ([raw file](https://github.com/metabase/metabase/blob/272192811add66c0895ee774556805edb82e1f0d/frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx#L100)):*
> ```JavaScript
> 
>     expect(
>       screen.getAllByTestId("tree-item-name").map(node => node.textContent),
> ```

What's the reason this line needs to be changed?

File: frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx [hunk 1]
```diff
@@ -1,96 +1,86 @@
...
-                        name={t.display_name}
+                        name={t.name}
                         icon={{
                           name: isDatasets ? "model" : "table2",
                           size: 16,
                         }}
-                        onSelect={() => onSelect(t)}
+                        onSelect={() => onSelect({ id: virtualTableId })}
                         rightIcon={PLUGIN_MODERATION.getStatusIcon(
                           t.moderated_status,
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionList.tsx:69]: Have you checked that `moderated_status` is returned by the search endpoint?

File: frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx [hunk 2]
```diff
@@ -52,38 +52,40 @@ function mockRootCollectionEndpoint(hasAccess) {
...
-      display_name: "a",
-      schema: requestedCollectionName,
-    }),
-    createMockTable({
-      id: "card__3",
-      display_name: "A",
-      schema: requestedCollectionName,
-    }),
-  ]);
+function mockCollectionItems() {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx:55]: There is `setupSearchEndpoints` helper

File: frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx [hunk 3]
```diff
@@ -104,10 +106,14 @@ describe("SavedQuestionPicker", () => {
   });
 
   it("sorts saved questions case-insensitive (metabase#23693)", async () => {
-    await setup();
+    const { fetchCollectionItemsMock } = await setup();
+
+    expect(fetchCollectionItemsMock.lastCall()[0]).toMatch(
+      "sort_column=name&sort_direction=asc",
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/saved-question-picker/SavedQuestionPicker.unit.spec.jsx:112]: I wouldn't add checks for the parameters here, it should be an implementation detail. What we could do instead is to make `setupSearchEndpoints` take these parameters into account, so here we would only assert for the correct text content (done below)

**Review by ranquild — COMMENTED** [review]
Some comments, mostly testing related

File: frontend/src/metabase/plugins/index.ts [hunk 4]
```diff
@@ -152,7 +151,7 @@ export const PLUGIN_MODERATION = {
   QuestionModerationButton: PluginPlaceholder,
   ModerationReviewBanner: PluginPlaceholder,
   ModerationStatusIcon: PluginPlaceholder,
-  getStatusIcon: object,
+  getStatusIcon: (moderated_status?: string) => undefined,
```
Comment (ranquild) [inline frontend/src/metabase/plugins/index.ts:154]: Could use use the correct type here?

**Review by ranquild — APPROVED** [review]

**Pull request merged**